From kay.sievers@vrfy.org  Sat Jul 18 15:43:32 2009
From: Kay Sievers <kay.sievers@vrfy.org>
Date: Sat, 18 Jul 2009 17:34:17 +0200
Subject: vc: fix vcs(a) devices for consoles
To: Greg KH <greg@kroah.com>
Cc: Gerardo Exequiel Pozzi <vmlinuz386@yahoo.com.ar>
Message-ID: <1247931257.29760.1.camel@yio.site>


From: Kay Sievers <kay.sievers@vrfy.org>

The buffer for the consoles are unconditionally allocated at
con_init() time, which miss the creation of the vcs(a) devices.

Since 2.6.30, these devices are no longer created at open()
and removed on close(), but controlled by the lifetime of the
buffers.

Reported-by: Gerardo Exequiel Pozzi <vmlinuz386@yahoo.com.ar>
Tested-by: Gerardo Exequiel Pozzi <vmlinuz386@yahoo.com.ar>
Cc: stable <stable@kernel.org>
Cc: Samuel Thibault <samuel.thibault@ens-lyon.org>
Cc: Nicolas Pitre <nico@cam.org>
Signed-off-by: Kay Sievers <kay.sievers@vrfy.org>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/char/vc_screen.c |    4 ++++
 1 file changed, 4 insertions(+)

--- a/drivers/char/vc_screen.c
+++ b/drivers/char/vc_screen.c
@@ -495,11 +495,15 @@ void vcs_remove_sysfs(int index)
 
 int __init vcs_init(void)
 {
+	unsigned int i;
+
 	if (register_chrdev(VCS_MAJOR, "vcs", &vcs_fops))
 		panic("unable to get major %d for vcs device", VCS_MAJOR);
 	vc_class = class_create(THIS_MODULE, "vc");
 
 	device_create(vc_class, NULL, MKDEV(VCS_MAJOR, 0), NULL, "vcs");
 	device_create(vc_class, NULL, MKDEV(VCS_MAJOR, 128), NULL, "vcsa");
+	for (i = 0; i < MIN_NR_CONSOLES; i++)
+		vcs_make_sysfs(i);
 	return 0;
 }
